code/game/gamemodes/changeling/powers/tiny_prick.dm

```diff
@@ -66,60 +66,44 @@
 	return 1
 
-/datum/action/changeling/sting/transformation
-	name = "Transformation Sting"
-	desc = "We silently sting a human, injecting a retrovirus that forces them to transform. Costs 50 chemicals."
-	helptext = "The victim will transform much like a changeling would. The effects will be obvious to the victim, and the process will damage our genomes."
-	button_icon_state = "sting_transform"
-	sting_icon = "sting_transform"
-	chemical_cost = 50
-	dna_cost = 3
-	genetic_damage = 100
-	var/datum/dna/selected_dna = null
-
-/datum/action/changeling/sting/transformation/Trigger()
-	var/mob/user = usr
-	var/datum/changeling/changeling = user.mind.changeling
-	if(changeling.chosen_sting)
-		unset_sting(user)
-		return
-	selected_dna = changeling.select_dna("Select the target DNA: ", "Target DNA")
-	if(!selected_dna)
-		return
-	if((NOTRANSSTING in selected_dna.species.species_traits) || selected_dna.species.is_small)
-		to_chat(user, "<span class='warning'>The selected DNA is incompatible with our sting.</span>")
-		return
-	..()
+/datum/action/changeling/sting/false_armblade //Port from TG, credit to them
+	name = "False Armblade Sting"
+	desc = "We silently sting a human, injecting a retrovirus that mutates their arm to temporarily appear as an armblade. Costs 20 chemicals."
+	helptext = "The victim will form an armblade much like a changeling would, except the armblade is dull and useless."
+	button_icon_state = "sting_armblade"
+	sting_icon = "sting_armblade"
+	chemical_cost = 20
+	dna_cost = 2
+	genetic_damage = 15
+	max_genetic_damage = 25
 
-/datum/action/changeling/sting/transformation/can_sting(var/mob/user, var/mob/target)
-	if(!..())
-		return
-	if((HUSK in target.mutations) || (!ishuman(target)))
-		to_chat(user, "<span class='warning'>Our sting appears ineffective against its DNA.</span>")
-		return FALSE
-	if(ishuman(target))
-		var/mob/living/carbon/human/H = target
-		if(NO_DNA in H.dna.species.species_traits)
-			to_chat(user, "<span class='warning'>This won't work on a creature without DNA.</span>")
-			return FALSE
-	return TRUE
+/obj/item/melee/arm_blade/false
+	desc = "A grotesque mass of flesh that used to be your arm. Although it looks dangerous at first, you can tell it's actually quite dull and useless."
+	force = 5 //Basically as strong as a punch
+	fake = TRUE
 
-/datum/action/changeling/sting/transformation/sting_action(var/mob/user, var/mob/target)
-	add_attack_logs(user, target, "Transformation sting (changeling) (new identity is [selected_dna.real_name])")
-	if(issmall(target))
-		to_chat(user, "<span class='notice'>Our genes cry out as we sting [target.name]!</span>")
+/datum/action/changeling/sting/false_armblade/sting_action(var/mob/user, var/mob/living/carbon/target)
+	add_attack_logs(user, target, "Fake armblade sting (changeling)")
 
-	if(iscarbon(target) && (target.status_flags & CANWEAKEN))
-		var/mob/living/carbon/C = target
-		C.do_jitter_animation(500)
+	if(!target.drop_item())
```

Comment:
It tries to drop active hand.
If we have a nodrop thing in active hand, and an empty inactive hand - this call will fail, and tell you "o no we can't grow a fake, because the user is currently more focused on their hand that's holding something"